backends/imgui_impl_vulkan.cpp

```diff
@@ -1870,9 +1880,11 @@ static void ImGui_ImplVulkan_SwapBuffers(ImGuiViewport* viewport, void*)
     info.pSwapchains = &wd->Swapchain;
     info.pImageIndices = &present_index;
     err = vkQueuePresentKHR(v->Queue, &info);
-    if (err == VK_ERROR_OUT_OF_DATE_KHR || err == VK_SUBOPTIMAL_KHR)
+    if (err == VK_ERROR_OUT_OF_DATE_KHR || err == VK_SUBOPTIMAL_KHR) {
         ImGui_ImplVulkanH_CreateOrResizeWindow(v->Instance, v->PhysicalDevice, v->Device, &vd->Window, v->QueueFamily, v->Allocator, (int)viewport->Size.x, (int)viewport->Size.y, v->MinImageCount);
-    else
+		return;
+	}
+	else
         check_vk_result(err);
```

Comment:
What's the value is returning here and not incrementing both indexes?

Reply:
Could you also confirm that both paths calling `ImGui_ImplVulkanH_CreateOrResizeWindow()` are needed?
Sorry for the amount of details requested, but since this is rather subtle and brittle and hard to reproduce, any extra details are useful for our records.

Reply:
none of this is needed but that's just how I handle it in my own backend. glad to see you figured it all